app/src/main/java/org/simple/clinic/teleconsultlog/shareprescription/TeleconsultSharePrescriptionModel.kt

```diff
@@ -52,4 +59,8 @@ data class TeleconsultSharePrescriptionModel(
   fun downloadButtonStateChanged(downloadButtonState: ButtonState?): TeleconsultSharePrescriptionModel {
     return copy(downloadButtonState = downloadButtonState)
   }
+
+  fun shareButtonStateChanged(shareButtonState: ShareButtonState?): TeleconsultSharePrescriptionModel {
```

Comment:
Can we have two separate methods like `sharing` and `sharingCompleted` to update the share button state.

Reply:
We are handling the button states using a single function across the project.

Reply:
same comment as download button state model function